packages/ptokens-assets-evm/src/lib/index.ts

```diff
@@ -44,84 +54,76 @@ export enum EVENT_NAMES {
 
 export const eventNameToSignatureMap = new Map<string, string>(
   events.map((_event) => {
-    const signature = encodeEventSignature(_event)
+    const signature = getEventSelector(_event)
     return [_event.name, signature]
   }),
 )
 
-const topicToAbiMap = new Map(
-  events.map((_event) => {
-    const signature = eventNameToSignatureMap.get(_event.name)
-    return [signature, _event as AbiEventFragment]
-  }),
-)
-
 const getOperationIdFromObj = (_obj: any) => {
-  const types = [
-    'bytes32',
-    'bytes32',
-    'bytes4',
-    'uint256',
-    'string',
-    'bytes4',
-    'string',
-    'string',
-    'uint256',
-    'address',
-    'bytes4',
-    'uint256',
-    'bytes',
-    'bytes32',
-  ]
-  return keccak256(
-    encodeParameters(types, [
-      _obj.originatingBlockHash || _obj.originBlockHash || _obj.blockHash,
-      _obj.originatingTransactionHash || _obj.originTransactionHash || _obj.transactionHash,
-      _obj.originatingNetworkId || _obj.originNetworkId || _obj.networkId,
-      _obj.nonce,
-      _obj.destinationAccount,
-      _obj.destinationNetworkId,
-      _obj.underlyingAssetName,
-      _obj.underlyingAssetSymbol,
-      _obj.underlyingAssetDecimals,
-      _obj.underlyingAssetTokenAddress,
-      _obj.underlyingAssetNetworkId,
-      _obj.assetAmount,
-      _obj.userData || '0x',
-      _obj.optionsMask,
-    ]),
+  const abiParameter = parseAbiParameter(
+    '(bytes32 originBlockHash, bytes32 originTransactionHash, bytes32 optionsMask, uint256 nonce, uint256 underlyingAssetDecimals, uint256 assetAmount, uint256 userDataProtocolFeeAssetAmount, uint256 networkFeeAssetAmount, uint256 forwardNetworkFeeAssetAmount, address underlyingAssetTokenAddress, bytes4 originNetworkId, bytes4 destinationNetworkId, bytes4 forwardDestinationNetworkId, bytes4 underlyingAssetNetworkId, string originAccount, string destinationAccount, string underlyingAssetName, string underlyingAssetSymbol, bytes userData, bool isForProtocol) operation',
```

Comment:
I would also check if it's possible the get the tuple/struct from the abi, maybe viem has this option.